src/awsexplorer/commands/copyName.ts

```diff
@@ -0,0 +1,28 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DefaultEnv, Env } from '../../shared/vscode/env'
+import { DefaultWindow, Window } from '../../shared/vscode/window'
+import { getLogger } from '../../shared/logger'
+import { AWSResourceNode } from '../../shared/treeview/nodes/awsResourceNode'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+
+/**
+ * Copies the name of the resource represented by the given node.
+ */
+export async function copyNameCommand(
+    node: AWSResourceNode,
+    window: Window = new DefaultWindow(),
+    env: Env = new DefaultEnv()
+): Promise<void> {
+    getLogger().debug(`CopyName called for ${node}`)
```

Comment:
Gave up on `toString`. Switched to

```typescript
getLogger().debug('CopyName called for %O', node)
```

> `%O: Object. A string representation of an object with generic JavaScript object formatting. Similar to util.inspect() without options. This will show the full object not including non-enumerable properties and proxies.`

Customized with [`[util.inspect.custom]`](https://nodejs.org/api/util.html#util_custom_inspection_functions_on_objects) to make the log messages more readable.

```typescript
public [inspect.custom](): string {
    return `S3FileNode (bucket=${this.bucket.name}, file=${this.file.key}}`
}
```